js/gameboard.js

```diff
@@ -1,110 +1,160 @@
-  return canvas;
-}
+/**
+ * This is a helper function that gets all of the search
+ * params in the current href given.
+ */
+window.location.getSearchParams = function (url) {
+  var search = url || this.search;
+  if (search.trim().length !== 0) {
+    search = search.slice(1);    
+    var splitSearchString = search.split('&');
 
-// places the image given on the canvas
-function imageInput (imageUrl, canvas){
-  // centext is 2 dimentional 
-  var context =  canvas.getContext('2d');
-  // new image
-  var myImage = getImage();
-  // upon image loading 
-  myImage.onload = function() {
-    // positioning and draw the image 
-    context.drawImage(myImage,0,0);
-  };  
-  // setting the empty image to an image 
-  myImage.src = imageUrl;
-  //prevents the image from stretching out 
-  myImage.width = "640px";
-  myImage.height = "480px";
-}
-// creating a variable that all the functions can get to
-var color;
-// just turn it red
-function setRed(){
- color = "#ff0000";
-}
-// just turn it green 
-function setGreen(){
-  color = "#009933";
+    var searchObject = {};
+    for (var i in splitSearchString) {
+      var str = splitSearchString[i].split('=')
+      var name = str[0];
+      var value = str[1];
+      searchObject[name] = strip(value, "%22");
+    }
+    return searchObject;
+  }
+  return undefined;
 }
-// just turn it blue
-function setBlue(){
-  color = "#0000ff";
-}
-// just turn it yellow
-function setYellow(){
-  color = "#ffff00";
+
+/**
+ * This gets the specfic search param in the href
+ */
+window.location.getSearchParam = function(name) {
+  return this.getSearchParams()[name];
 }
-// x-axis and y-axis; draws circle at that position
-function drawCircle(x, y){
-  // if x & y doesnt exist, get out of the function
-  if (! x || ! y){
-    console.log("Don't know where to go.");
-    return false;
-  } 
-  // canvas into the variable target
-  var target = getCanvas();
-  // assinging 2d to context 
-  var context = target.getContext("2d");
-  // shape that your drawing
-  context.beginPath();
-  var radius = 5; // the size is the radius in pixels    
-  var start = 0; 
-  var end = Math.PI*2;
-  context.arc(x, y, radius, start, end);
-  //calling the varible to the colors
-  context.fillStyle = color;
-  // filling the circles 
-  context.fill();
+
+/**
+ * This is the canvase class that is just a wrapper/refactor of the previous code base.
+ */
+function canvas() {
+  // the canvas element on the web page
+  this.canvasContext = $('#gameboard').get(0).getContext("2d");
+  
+  // the canvas's element offset
+  this._canvasOffet = $('#gameboard').offset();
+  
+  // the image being given by google MAP API in the href
+  this.imageUrl = strip(location.search.slice(1).split('screenshot')[1], '%22').slice(1);
+
+  // the color the paint brush is using
+  this.color = '';
+  
+  // some url
+  this.brickwall = "https://s-media-cache-ak0.pinimg.com/736x/94/aa/66/94aa6684bdc152457d6983e58fd24bd2.jpg";
```

Comment:
is this needed?

Reply:
No but was not sure if that something you need later but I'll take that out